game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProPurchaseOption.java

```diff
@@ -71,6 +72,10 @@
     final Resource pus = data.getResourceList().getResource(Constants.PUS);
     cost = productionRule.getCosts().getInt(pus);
     costs = productionRule.getCosts();
+    isConstruction = unitAttachment.getIsConstruction();
```

Comment:
These 4 data members seem very related. Have you considered creating a type around it?

Reply:
Not really. I'd have to think more about this class in general to see if/how it would be best to start breaking it up a bit. Compared to most AI classes though this is probably one of the better ones.